transport/errors.go

```diff
@@ -207,3 +210,32 @@ func (e procedureFailedError) Error() string {
 func (e procedureFailedError) AsHandlerError() HandlerError {
 	return LocalUnexpectedError(e)
 }
+
+//////////////////////////////////////////////////////////////////////////////
+
+type TimeoutError interface {
+	error
+
+	timeoutError()
+}
+
+type timeoutError struct {
+	Service   string
+	Procedure string
+	Duration  time.Duration
+}
+
+func NewTimeoutError(Service string, Procedure string, Duration time.Duration) timeoutError {
+	return timeoutError{
+		Service:   Service,
+		Procedure: Procedure,
+		Duration:  Duration,
+	}
+}
+
+func (e timeoutError) timeoutError() {}
+
+func (e timeoutError) Error() string {
+	return fmt.Sprintf(`timeout for procedure %q of service %q after %v`,
+		e.Procedure, e.Service, e.Duration)
```

Comment:
yeah `%v` formats to include the duration so this should look fine (like `1s, 389.5ms` etc)